Kuadrant/limitador-operator
# local catalog raw file based format

### what

* Locally built custom OLM catalogs use (raw) File Based Catalog format instead of the deprecated SQLITE format.
From [SQLite-based catalogs](https://docs.openshift.com/container-platform/4.9/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-sqlite),

```
The SQLite database format for Operator catalogs is a deprecated feature. Deprecated functionality is still included in OpenShift Container Platform and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.
```

* Update github actions to push file based catalog format images
* Makefile
  * target renamed: `deploy-olm` -> `deploy-catalog`
  * target renamed: `undeploy-olm` -> `undeploy-catalog`
  * function renamed: `go-get-tool` -> `go-install-tool`
  * `bundle-dependencies` target removed. Functionality moved to the `bundle` target.
  * replace `envsubst` operations to update yaml objects with external env variarbles with the `yq` tool
  * moved all catalog related logic to `make/catalog.mk`
  * New: `catalog` target to build catalog manifests
* Documentation oriented to contributors `doc/development.md`:
  * New section: `Deploy kuadrant operator using OLM`
  * New section: "Build custom OLM catalog`
  * Added Table of Contents.

### Verification steps

Build local catalog. It will include kuadrant operator, limitador operator and authorino operator

```
make catalog
make catalog-build CATALOG_IMG=quay.io/kuadrant/limitador-operator-catalog:local-testing
```

Push the image to Quay.io. Currently it is required because the catalogsource sets the image pull policy to `Always`, so even if the image is loaded in the kind cluster, it will fail if the image is not in quay.

```
make catalog-push CATALOG_IMG=quay.io/kuadrant/limitador-operator-catalog:local-testing
```

Create kind cluster

```
make kind-create-cluster
```

Deploy OLM

```
make install-olm
```

Deploy kuadrant operator using OLM

```
make deploy-catalog CATALOG_IMG=quay.io/kuadrant/limitador-operator-catalog:local-testing
```

Check the limitador operator is up and running (it takes up to few minutes)

```
k get pods -n limitador-system
NAME                                                              READY   STATUS      RESTARTS   AGE
NAME                                                              READY   STATUS      RESTARTS   AGE
6427bb5aac2edc4750e505a33be4eb4b56923d3aed4213abecdc70--1-l4vgf   0/1     Completed   0          10s
limitador-operator-catalog-2p8bg                                  1/1     Running     0          54s
```
